pkg/internal/config/config.go

```diff
@@ -120,64 +120,88 @@ func New(baseConfig kconfig.Config) (Config, error) {
 	}
 
-	if rawTrafficGeneratorEastMacAddress :=
-		baseConfig.Params[TrafficGeneratorEastMacAddressParamName]; rawTrafficGeneratorEastMacAddress != "" {
-		trafficGeneratorEastMacAddress, err := net.ParseMAC(rawTrafficGeneratorEastMacAddress)
+	if rawVal := baseConfig.Params[TrafficGeneratorEastMacAddressParamName]; rawVal != "" {
+		newConfig.TrafficGeneratorEastMacAddress, err = parseMacAddress(rawVal, ErrInvalidTrafficGeneratorEastMacAddress)
 		if err != nil {
-			return Config{}, ErrInvalidTrafficGeneratorEastMacAddress
+			return Config{}, err
 		}
-		newConfig.TrafficGeneratorEastMacAddress = trafficGeneratorEastMacAddress
 	}
 
-	if rawTrafficGeneratorWestMacAddress :=
-		baseConfig.Params[TrafficGeneratorWestMacAddressParamName]; rawTrafficGeneratorWestMacAddress != "" {
-		trafficGeneratorWestMacAddress, err := net.ParseMAC(rawTrafficGeneratorWestMacAddress)
+	if rawVal := baseConfig.Params[TrafficGeneratorWestMacAddressParamName]; rawVal != "" {
+		newConfig.TrafficGeneratorWestMacAddress, err = parseMacAddress(rawVal, ErrInvalidTrafficGeneratorWestMacAddress)
 		if err != nil {
-			return Config{}, ErrInvalidTrafficGeneratorWestMacAddress
+			return Config{}, err
 		}
-		newConfig.TrafficGeneratorWestMacAddress = trafficGeneratorWestMacAddress
 	}
 
-	if rawEastDPDKMacAddress := baseConfig.Params[DPDKEastMacAddressParamName]; rawEastDPDKMacAddress != "" {
-		dpdkEastMacAddress, err := net.ParseMAC(rawEastDPDKMacAddress)
+	if rawVal := baseConfig.Params[DPDKEastMacAddressParamName]; rawVal != "" {
+		newConfig.DPDKEastMacAddress, err = parseMacAddress(rawVal, ErrInvalidDPDKEastMacAddress)
 		if err != nil {
-			return Config{}, ErrInvalidDPDKEastMacAddress
+			return Config{}, err
 		}
-		newConfig.DPDKEastMacAddress = dpdkEastMacAddress
 	}
 
-	if rawWestDPDKMacAddress := baseConfig.Params[DPDKWestMacAddressParamName]; rawWestDPDKMacAddress != "" {
-		dpdkWestMacAddress, err := net.ParseMAC(rawWestDPDKMacAddress)
+	if rawVal := baseConfig.Params[DPDKWestMacAddressParamName]; rawVal != "" {
+		newConfig.DPDKWestMacAddress, err = parseMacAddress(rawVal, ErrInvalidDPDKWestMacAddress)
 		if err != nil {
-			return Config{}, ErrInvalidDPDKWestMacAddress
+			return Config{}, err
 		}
-		newConfig.DPDKWestMacAddress = dpdkWestMacAddress
 	}
 
-	if rawTestDuration := baseConfig.Params[TestDurationParamName]; rawTestDuration != "" {
-		testDuration, err := time.ParseDuration(rawTestDuration)
+	if rawVal := baseConfig.Params[TestDurationParamName]; rawVal != "" {
+		newConfig.TestDuration, err = parseTestDuration(rawVal)
 		if err != nil {
-			return Config{}, ErrInvalidTestDuration
+			return Config{}, err
 		}
-		newConfig.TestDuration = testDuration
 	}
 
 	return newConfig, nil
 }
+
+func parseTrafficGeneratorPacketsPerSecondInMillions(rawVal string) (int, error) {
+	val, err := strconv.Atoi(rawVal)
+	if err != nil || val < 0 {
+		return 0, ErrInvalidTrafficGeneratorPacketsPerSecondInMillions
+	}
+	return val, nil
+}
+
```

Comment:
Please consider generalizing this function to `parseNonZeroPositiveInt(rawVal string) (int, error)`.
Or something along those lines.

Reply:
Done